src/components/generic/NumberStepper/NumberStepper.tsx

```diff
@@ -0,0 +1,127 @@
+import type React from "react";
+import type { ReactNode } from "react";
+import { useState, useEffect, useRef } from "react";
+import styles from "./NumberStepper.module.scss";
+import { formatTimerNumber } from "../../../utils/helpers.tsx";
+
+interface NumberStepperProps {
+    label?: string | ReactNode;
+    value: number; // Current value of the stepper
+    onChange: (newValue: number) => void; // Function to update the value
+    min?: number; // Minimum value
+    max?: number; // Maximum value
+    step?: number; // Increment/decrement step
+}
+
+const NumberStepper: React.FC<NumberStepperProps> = ({
+                                                         value,
+                                                         onChange,
+                                                         min = Number.NEGATIVE_INFINITY,
+                                                         max = Number.POSITIVE_INFINITY,
+                                                         step = 1,
+                                                         label,
+                                                     }) => {
+    const [isEditing, setIsEditing] = useState(false); // State to toggle between input and display mode
+    const [localValue, setLocalValue] = useState(value); // Local state for handling the current value
+    const inputRef = useRef<HTMLInputElement>(null); // Ref for the input field
+
+    useEffect(() => {
+        setLocalValue(value); // Keep local state in sync with the external value
```

Comment:
not sure why you're keeping state in two places. I could see this being useful for debounced inputs where the outer value is slightly delayed and not all updates are reported externally, but here there's no reason to keep two states.